homework2.py

```diff
@@ -0,0 +1,428 @@
+	>>> 'black-green-red-white-yellow'
+
+	Arguments:
+	hyphen_str: input string separated by hyphen
+
+	Returns:
+	sorted_str: string in a hyphen-separated sequence after
+	sorting them alphabetically
+'''
+
+def sort_words(hypehn_string):
+    hypehn_string=[n for n in hypehn_string.split('-')]
+    hypehn_string.sort()
+    return("The alphabetically sorted words are:" + '-'.join(hypehn_string))
+
+sort_words('green-red-yellow-black-white')
+
+'''
+4.def perfect_number():
+''''''
+	Write a Python function to check whether a number is perfect or not.
+
+	A perfect number is a positive integer that is equal to the sum of
+	its proper positive divisors. Equivalently, a perfect number is a number
+	that is half the sum of all of its positive divisors (including itself).
+
+	Example: 6 is a perfect number as 1+2+3=6. Also by the second definition,
+	(1+2+3+6)/2=6. Next perfect number is 28=1+2+4+7+14. Next two perfect
+	numbers are 496 and 8128.
+
+	Argument:
+	number: number to check
+
+	Returns:
+	perfect: boolean, True if number is perfect
+''''
+
+def perfect_number(a):
+    sum1=0
+    for i in range(1, a):
+        if(a%i==0):
+            sum1=sum1+i
+    if sum1==a:
+        return True
+    else:
+        return False
+
+perfect_number(8128)
+
+
+if __name__ == '__main__':
+	pass
+
+
+''''
+Edit this file to complete Exercise 7
+'''
+
+import csv
+import json
+import os
+import os.path
+from os import path
+from pathlib import Path
+# import the modules you need here
+
+'''
+1.def check_path(path):
+	''''''
+	check if the input path exists, if it exists, return True and a
+	list containing the following
+		1. if this path is absolute path
+		2. if this path is a directory
+		3. if this path is a file
+	if input path doesn't exist, return False and empty list
+
+	Arguments:
+	path: input path as a string
+
+	Returns:
+	exist_flag: True if path exist, False otherwise
+	path_info_list: list containing 3 boolean varaibles for 3 checks
+		performed if path exists, return empty list if path doesn't
+		exist.
+		''''
+def check_path(xyz):
+    list123=[]
+    if path.exists(xyz) == True:
+        list123.append(path.isabs(xyz))
+        list123.append(path.isfile(xyz))
+        list123.append(path.isdir(xyz))
+    print(str(path.exists(xyz)) + ',' + str (list123))
+
+	check_path('/Users/sun/Desktop/ds-class-intro')
+
+	'''
+	Example:
+	check_path('some/path/of/a/directory')
+	# if it exists
+	>>> True, [False, True, False]
+
+	# if doesn't exist
+	>>> False, []
+	'''
+
+
+
+'''       ****************************Question Mark***********************************
+2.def read_csv(file):
+	''''''
+	reads in a csv file then return the total number of lines in it
+
+	Arguments:
+	file: path to the file to read
+
+	Returns:
+	Number of rows in the input file
+
+	Example:
+	read_csv('AMZN.csv')
+	>>> 14
+	'''
+
+'''There is something wrong with this question, but I cannot figure out	'''
+def read_csv1(file):
+    open(file,'r')
```

Comment:
you're not using `open` correctly... check out our notebook 03 where we discussed how to open and read file.

Reply:
looks like you're using `open` correctly in the following problem... you should be able to fix it